include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -139,6 +139,85 @@ class zip_forward_iterator
     __it_types __my_it_;
 };
 
+template <typename T, typename _UnaryFunc>
+auto
+make_transform_output_ref_wrapper(T&& __reference, _UnaryFunc __unary_func);
+
+template <typename T, typename _UnaryFunc>
+class transform_output_ref_wrapper
+{
+  private:
+    T& __my_reference_;
+    _UnaryFunc __my_unary_func_;
+
+  public:
+    typedef ::std::true_type IsTransformOutputIterRefWrapper;
+    transform_output_ref_wrapper(T& __reference, _UnaryFunc __unary_func)
+        : __my_reference_(__reference), __my_unary_func_(__unary_func)
+    {
+    }
+
+    operator T&() { return __my_reference_; }
+
+    template <typename _UnaryFuncOuter>
+    auto
+    make_composite_wrapper(_UnaryFuncOuter __unary_func_outer)
+    {
+        return make_transform_output_ref_wrapper(__my_reference_,
+                                                 [=](auto x) { return __my_unary_func_(__unary_func_outer(x)); });
```

Comment:
Is it ok that data copied here?
I mean "auto x"

Reply:
No, this should be `const auto& x`.  Fixed.